internal/cli/kraft/cloud/service/create/create.go

```diff
@@ -58,7 +58,7 @@ func Create(ctx context.Context, opts *CreateOptions, args ...string) (*kraftclo
 		)
 	}
 
-	services := []kraftcloudservices.Service{}
+	services := make([]kcservices.CreateRequestService, 0, 2)
```

Comment:
Is this correct? A service can have more than 2 open ports

Reply:
It's a naive pre-allocation for the case where the user did not provide service group details. `append` will automatically grow the array if we add more than 2 elements.

As far as I can tell this is the only case where we can pre-allocate without parsing. However we could also just allocation within each `if` case.

Reply:
I split the allocations into the two cases in [da0715e8..98db95e3](https://github.com/unikraft/kraftkit/compare/da0715e8aa67c125930667f42ef9633f0cb295ed..98db95e38fed65aa0fbfbaf7fddd20f5f08d35cd). I agree that this feel clearer.